packages/fx-core/src/plugins/resource/localdebug/index.ts

```diff
@@ -201,6 +204,38 @@ export class LocalDebugPlugin implements Plugin {
             ctx.config.set(LocalDebugConfigKeys.SkipNgrok, "false");
             ctx.config.set(LocalDebugConfigKeys.LocalBotEndpoint, "");
           }
+        } else {
+          // add 'npm install' scripts into root package.json
+          const packageJsonPath = ctx.root;
+          let packageJson: any = undefined;
+          try {
+            packageJson = await PackageJson.load(packageJsonPath);
+          } catch (error) {
+            ctx.logProvider?.error(`Cannot load package.json from ${ctx.root}. ${error}`);
+          }
+
+          if (packageJson !== undefined) {
+            const scripts = packageJson.content.scripts ?? {};
+            const installAll: string[] = [];
+
+            if (includeBackend) {
```

Comment:
Maybe it is also necessary to check if tabs/api/bot folder exist?

Reply:
This is to change root package.json so nothing to do with tabs/api/bot folder.
BTW, plugin scaffolding may happen concurrently so tabs/api/bot folder may not be created yet. The only flag to use is `includeXxx`.
That's also the same logic as generating `launch.json` and `tasks.json`.

Reply:
My mistake.